src/main/java/com/teragrep/hbs_03/hbase/MetaRow.java

```diff
@@ -0,0 +1,126 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.hbs_03.hbase;
+
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.jooq.Record21;
+import org.jooq.types.UInteger;
+import org.jooq.types.ULong;
+
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.util.HashMap;
+import java.util.Map;
+
+/** Represents a row for meta-column family */
+public final class MetaRow implements Row {
+
+    private final Record21<ULong, Date, Date, String, String, String, String, String, Timestamp, ULong, String, String, String, String, String, String, ULong, UInteger, String, String, Long> record;
+    private final Binary rowKey;
+
+    public MetaRow(
+            final Record21<ULong, Date, Date, String, String, String, String, String, Timestamp, ULong, String, String, String, String, String, String, ULong, UInteger, String, String, Long> record
+    ) {
+        this(record, new MetaRowKey(record.value18().longValue(), record.value21(), record.value1().longValue()));
+    }
+
+    public MetaRow(
+            final Record21<ULong, Date, Date, String, String, String, String, String, Timestamp, ULong, String, String, String, String, String, String, ULong, UInteger, String, String, Long> record,
+            final Binary rowKey
+    ) {
+        this.record = record;
+        this.rowKey = rowKey;
+    }
+
+    public Put put() {
```

Comment:
To get rid of instanceof -operator usage in ObjectToBinary class that I mentioned in my review, refactoring is also needed here.
One way to do this is to remove the map object generation completely and instead replace the map.put code lines with methods that directly add the familyBytes, qualifierBytes and record field to the Put object.

i.e.
`put.addColumn(familyBytes, Bytes.toBytes("i"), new StringToBinary(record.field1().get(record)).bytes());`
Where StringToBinary.java is a class for generating Binary object from the Strings, along with the hbase NULL value handling.